### DISCUSSION TITLE
Date Tokens Question

### DISCUSSION THREAD

**Discussion by dcashpeterson** [microsoft-search/pnp-modern-search#2623] [Q&A] [answered]
Hello all, 
Not sure if this is a bug or PEBCAK. I'm trying to search based on all the items this month like this. AnniversaryOWSDATE:2000-{CurrentMonth}

This works great for months with two charters 10-12 but for January through September the token comes back without the 0 in front so you get AnniversaryOWSDATE:2000-1 instead of AnniversaryOWSDATE:2000-01 which returns no results.

What is the best way to do this?

**Comment by wobba** [comment, marked as answer]
`AnniversaryOWSDATE` is a string property. Since we don't zero pad the variables as that is not needed when you use date properties you could try this:

`AnniversaryOWSDATE:{CurrentYear}-0{CurrentMonth} OR AnniversaryOWSDATE:{CurrentYear}-{CurrentMonth}` which would expand to:

`AnniversaryOWSDATE:2022-01 OR AnniversaryOWSDATE:2022-1` for double digit month it would be `2022-011 OR 2022-11` so one should always match.

**Comment by dcashpeterson** [comment]
@wobba Thank you. It was PEBCAK. :)

**Reply by wobba** [reply]
Well... but not super intuitive either 🙂
